packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -558,6 +569,18 @@ class PipelineController extends CanvasController {
     if (pipeline !== undefined) {
       const app_data = prefixedToNested(data);
       this.setNodeProperties(nodeID, { app_data }, pipeline.id);
+      if (data.label !== data.ui_data?.label) {
```

Comment:
I feel like an "escape" might be warranted in the case where `data.label === ""` (aka the default). That case will have this statement run, but I'm unsure if it would be redundant

Reply:
When label is an empty string, we want it to be set to the default label, so it should probably also run in that case